### DISCUSSION TITLE
Conan private registry support

### DISCUSSION THREAD

**Discussion by timmay54** [renovatebot/renovate#24092] [Suggest an Idea]
### Tell us more.

What would you like Renovate to be able to do?
Support private registries of Conan packages to search/scan for when evaluating potential updates.

Is this a feature you are interested in implementing yourself?
Yes

**Comment by timmay54** [comment]
Using the private registry support that pip has, I would imagine its something that can be ported to be used by conan (looking to implement support for private registries that are hosted by gitlab self-hosted and artifactory self-hosted)

**Comment by rarkins** [comment]
How is authentication done for Conan private registries? If it's standardized authentication (e.g. Bearer or Basic token) then it might be possible without any code changes

**Reply by timmay54** [reply]
it is performed with an API token from the private registry; so as long as `conan user -r homelab -p someApiTokenAuth timmay54` has ran, then renovate should be able to do a "conan search" against all authed remotes

**Reply by rarkins** [reply]
We need to know how it works for the http requests (bearer or basic, etc). An example curl request is usually enough

**Reply by rarkins** [reply]
And we don't and won't run any "Conan x" commands - we perform lookups directly.

**Reply by timmay54** [reply]
Ah gotchya - its Bearer; Looking at how conan is fetching information from the gitlab api, it is using a gitlab API token to auth a GET request to 
`{servername}/api/v4/projects/{projectID}/packages`. 

Curl --header "private-token:TOKEN" https://my.gitlab.tv/api/v4/555/packages/conan

More info here: https://docs.gitlab.com/ee/api/packages.html
and here: https://docs.gitlab.com/ee/api/packages/conan.html

**Reply by rarkins** [reply]
What URL does _Renovate_ use today though? There's no point talking credentials if the URL is wrong in the first place.

Secondly, `private-token:` is a proprietary header, this is not standardized Bearer auth
